src/Features/Core/Portable/Diagnostics/BuildOnlyDiagnosticsService.cs

```diff
@@ -51,62 +70,68 @@ private void OnWorkspaceChanged(object? sender, WorkspaceChangeEventArgs e)
                 case WorkspaceChangeKind.AdditionalDocumentReloaded:
                 case WorkspaceChangeKind.AnalyzerConfigDocumentRemoved:
                 case WorkspaceChangeKind.AnalyzerConfigDocumentReloaded:
-                    ClearDiagnostics(e.DocumentId);
+                    _workQueue.AddWork(cancellationToken => ClearDiagnosticsAsync(e.DocumentId, cancellationToken));
```

Comment:
workspace events just go intot eh queue and are processed when teh queue operates.